src/admin/controller/modules/admin-ops.xqy

```diff
@@ -9,11 +9,19 @@ declare function admin-ops:document-insert(
 )
 as empty-sequence()
 {
-  xdmp:document-insert(
-    $uri,
-    $doc,
-    xdmp:default-permissions()
-  )
+  if (fn:doc-available($uri)) then
```

Comment:
I think you can just do this. No need for the else. xdmp:document-get-collections('bogus') will return ()

```xquery
xdmp:document-insert(
  $uri,
  $doc,
  xdmp:document-get-permissions($uri),
  xdmp:document-get-collections($uri)
)
```